constrain/api/workflow.py

```diff
@@ -47,6 +46,46 @@ def __init__(self, workflow, run_workflow_now=False):
 
         self.load_states()
 
+        # First, check if the workflow_dict has "working_dir"
```

Comment:
Please put the whole working dir logic into a dedicated method

Reply:
I have moved this whole working dir logic to a dedicated method.